src/core/utils/color/ColorValue.ts

```diff
@@ -0,0 +1,709 @@
+    return tbgr === this._tbgr ? this : ColorValue.fromTbgr(tbgr);
+  }
+
+  /** Compute the 0xTTBBGGRR value of the specified color and transparency.
+   * @param transparency the new transparency as an integer between 0-255.
+   * @returns The 0xTTBBGGRR value equivalent to `tbgr` but with the specified transparency.
+   */
+  public static withTransparency(tbgr: number, transparency: number): number {
+    return this.withAlpha(tbgr, 255 - transparency);
+  }
+
+  /** The "known name" for this ColorValue. Will be undefined if color value is not in [[ColorByName]] list */
+  public get name(): string | undefined {
+    return ColorValue.getName(this.tbgr);
+  }
+
+  /** Obtain the name of the color in the [[ColorByName]] list associated with the specified 0xTTBBGGRR value, or undefined if no such named color exists. */
+  public static getName(tbgr: number): string | undefined {
+    return ColorByName[tbgr];
+  }
+
+  /** Convert this ColorValue to a string in the form "#rrggbb" where values are hex digits of the respective colors */
+  public toHexString(): string {
+    return ColorValue.toHexString(this.tbgr);
+  }
+
+  /** Convert the 0xTTBBGGRR value to a string in the form "#rrggbb". */
+  public static toHexString(tbgr: number): string {
+    return `#${`000000${this.getRgb(tbgr).toString(16)}`.slice(-6)}`;
+  }
+
+  private static getColorsString(tbgr: number) {
+    const c = this.getColors(tbgr);
+    return `${c.r}, ${c.g}, ${c.b}`;
+  }
+
+  /** Convert this ColorValue to a string in the form "rgb(r,g,b)" where values are decimal digits of the respective colors. */
+  public toRgbString(): string {
+    return ColorValue.toRgbString(this.tbgr);
+  }
+
+  /** Convert the 0xTTBBGGRR color to a string in the form "rgb(r,g,b)" where each component is specified in decimal. */
+  public static toRgbString(tbgr: number): string {
+    return `rgb(${this.getColorsString(tbgr)})`;
+  }
+
+  /** Convert this ColorValue to a string in the form "rgba(r,g,b,a)" where color values are decimal digits and a is a fraction */
+  public toRgbaString(): string {
+    return ColorValue.toRgbaString(this.tbgr);
+  }
+
+  /** Convert the 0xTTBBGGRR color to a string of the form "rgba(r,g,b,a)" where the color components are specified in decimal and the alpha component is a fraction. */
+  public static toRgbaString(tbgr: number): string {
+    return `rgba(${this.getColorsString(tbgr)}, ${(
+      this.getAlpha(tbgr) / 255
+    ).toFixed(2)})`;
+  }
+
+  /** Create a ColorValue from hue, saturation, lightness values. transparency (0-255)  */
+  public static fromHSL(hsl: HslColor, transparency = 0): ColorValue {
+    return this.fromTbgr(
+      this.computeTbgrFromHSL(
+        hsl.h / 360,
+        hsl.s / 100,
+        hsl.l / 100,
+        transparency,
+      ),
+    );
+  }
+
+  /** Compute the 0xTTBBGGRR color corresponding to the specified hue, saturation, lightness values. */
+  public static computeTbgrFromHSL(
```

Comment:
Wouldnt it simplify, if we just ask to create `ColorValue` instance from what ever color input and then have method `toTbgr`? Similar as with other color schemes